wdl-engine/src/stdlib/read_tsv.rs

```diff
@@ -0,0 +1,480 @@
+//! Implements the `read_tsv` function from the WDL standard library.
+
+use std::fs;
+use std::io::BufRead;
+use std::io::BufReader;
+use std::sync::Arc;
+
+use anyhow::Context;
+use indexmap::IndexMap;
+use itertools::Either;
+use itertools::EitherOrBoth;
+use itertools::Itertools;
+use wdl_analysis::stdlib::STDLIB as ANALYSIS_STDLIB;
+use wdl_analysis::types::PrimitiveTypeKind;
+use wdl_ast::Diagnostic;
+use wdl_grammar::lexer::v1::is_ident;
+
+use super::CallContext;
+use super::Function;
+use super::Signature;
+use crate::Array;
+use crate::CompoundValue;
+use crate::PrimitiveValue;
+use crate::Value;
+use crate::diagnostics::function_call_failed;
+
+/// Represents a header in a TSV (tab-separated value) file.
+enum TsvHeader {
+    /// The header was explicitly specified as an `Array[String]`.
+    Specified(Array),
+    /// The header was read from the file.
+    File(String),
+}
+
+impl TsvHeader {
+    /// Gets the column names in the header.
+    ///
+    /// # Panics
+    ///
+    /// Panics if a specified header contains a value that is not a string.
+    pub fn columns(&self) -> impl Iterator<Item = &str> {
+        match self {
+            Self::Specified(array) => Either::Left(array.elements().iter().map(|v| {
+                v.as_string()
+                    .expect("header value must be a string")
+                    .as_str()
+            })),
+            Self::File(s) => Either::Right(s.split('\t')),
+        }
+    }
+}
+
+/// Reads a tab-separated value (TSV) file as an Array[Array[String]]
+/// representing a table of values.
+///
+/// Trailing end-of-line characters (\r and \n) are removed from each line.
+///
+/// `Array[Array[String]] read_tsv(File, [false])``: Returns each row of the
```

Comment:
I don't understand this signature. What's `[false]` mean? And looks like an extra backtick.

Reply:
There's a couple of extra backticks in comments that VS Code auto inserted for me that I'll fix.

Regarding the signature, that's a great question. It takes a `Boolean` here, but it's _required_ to be `false` for this overload, which means this function is actually overloaded on its _return type_ only, which is incorrect.

I filed https://github.com/openwdl/wdl/issues/690 earlier to ask them to clean this up.